src/core/core-varlink.c

```diff
@@ -512,34 +530,47 @@ int manager_setup_varlink_server(Manager *m) {
         if (m->varlink_server)
                 return 0;
 
-        if (!MANAGER_IS_SYSTEM(m))
-                return -EINVAL;
```

Comment:
Please remove the MANAGER_IS_SYSTEM() condition on varlink server deserialization too.